tooling/template-base/vite.config.ts

```
@@ -1,7 +1,20 @@
-import { tiConfig } from '@thoughtindustries/helium-server';
 import { defineConfig } from 'vite';
+import react from '@vitejs/plugin-react';
+import path from 'path';
 
 export default defineConfig(async () => {
+  const plugins = [react()];
+
+  try {
+    // Try to load vike dynamically
+    const vikeModule = await import('vike/plugin');
```

Is there a reason to load vike dynamically here vs importing it from the start? Seems like it could fail silently and SSR would break -- so it'd look like it was working, when it's really limping since it'd be hard to see with it just warning. Feels like something we should surface?

---

New Implementation:

```node
// In helium-server/src/vite-config/vite.config.ts
export async function createVikeConfig() {
  try {
    const vikeModule = await import('vike/plugin');
    const vike = vikeModule.default || vikeModule;
    
    return mergeConfig(baseConfig, {
      plugins: [vike()]
    });
  } catch (error) {
    console.error('Failed to load vike plugin:', error);
    throw new Error('Vike plugin is required but could not be loaded. Please ensure vike is installed.');
  }
}
```

**Why dynamic import:** We encountered ESM/CJS compatibility issues where Node.js was trying to `require()` an ESM-only module (vike), causing "ERR_REQUIRE_ESM" errors

**Addresses silent failure concern:** We throw an explicit error that stops the build/dev server if vike fails to load

**Clear error messaging:** The error message clearly states what's wrong and what's needed

**No Partial Functionality:** The application won't start without vike, ensuring no partial functionality

These ESM/CJS compatibility issues are really tricky and finicky especially with Vike so this was the best solution I could come up with that didn't break everything.